script.js

```diff
@@ -1,4 +1,84 @@
+var winCount = 0;
+var numDraws = 0;
+var lossCount = 0;
+var currentMode = "waiting for user name";
+var userName = "";
+var myOutputValue = "";
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (currentMode == "waiting for user name") {
+    // set the name of user
+    userName = input;
+
+    currentMode = "game mode";
+    myOutputValue = `Your name is ${userName}.`;
+    return myOutputValue;
+  } else if (currentMode == "game mode") {
+    var spsGenerator = spsOptions();
+    console.log(spsGenerator);
+    var option = "";
+
+    //if computer rolls 1, it will show paper
+    if (spsGenerator == 1) {
+      option = "paper";
+    }
+    //if computer rolls 2, it will show scissors
+    if (spsGenerator == 2) {
+      option = "scissors";
+    }
+    //if computer rolls 3, it will show stone
+    if (spsGenerator == 3) {
+      option = "stone";
+    }
+
```

Comment:
Can consider this bit being inside of `spsOptions()`, that way the helper function returns either 'scissors'/'paper'/'stone' instead of a number, and we can keep `main()` neat and tidy :)